services/vision/detectionstosegments/detections_to_3dsegments.go

```diff
@@ -0,0 +1,67 @@
+// Package detectionstosegments uses a 2D segmenter and a camera that can project its images
+// to 3D to project the bounding boxes to 3D in order to created a segmented point cloud.
+package detectionstosegments
+
+import (
+	"context"
+	"image"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/utils"
+	"go.viam.com/rdk/vision/objectdetection"
+	"go.viam.com/rdk/vision/segmentation"
+)
+
+var model = resource.NewDefaultModel("detector_3d_segmenter")
+
+func init() {
+	resource.RegisterService(vision.Subtype, model, resource.Registration[vision.Service, *segmentation.DetectionSegmenterConfig]{
+		DeprecatedRobotConstructor: func(ctx context.Context, r any, c resource.Config, logger golog.Logger) (vision.Service, error) {
+			attrs, err := resource.NativeConfig[*segmentation.DetectionSegmenterConfig](c)
+			if err != nil {
+				return nil, err
+			}
+			actualR, err := utils.AssertType[robot.Robot](r)
+			if err != nil {
+				return nil, err
+			}
+			return register3DSegmenterFromDetector(ctx, c.ResourceName().Name, attrs, actualR)
+		},
+	})
+}
+
+// register3DSegmenterFromDetector creates a 3D segmenter from a previously registered detector.
+func register3DSegmenterFromDetector(
```

Comment:
[q] So are these services fine to just always rebuild? Based on the looks of this, the vision service itself wont know how to reconfigure this type of "sub-service"

Reply:
fine for AlwaysRebuild